app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -111,4 +112,15 @@ class MainActivity : AppCompatActivity() {
                 .replace(R.id.frameContainer, fragment)
                 .commit()
     }
+
+    override fun onBackPressed() {
+        val currentFragmentClass = this.supportFragmentManager.findFragmentById(R.id.frameContainer).javaClass
+        if (currentFragmentClass != EventsFragment().javaClass) {
```

Comment:
Check using `instanceOf` 

Secondly, you are instantiating EventsFragment here - `EventsFragment().javaClass` Why?

Reply:
My question is that why are you instantiating the fragment? Why are `()` after the class name?

Nevertheless, this won't be needed when you use `instanceOf`

Reply:
Read about static fields of a class. However, implement this by `is` keyword of Kotlin